sdk_contrib/fetch/lib/subsegment_fetch.js

```diff
@@ -0,0 +1,37 @@
+const {Subsegment} = require('aws-xray-sdk-core');
+
+/**
+ * Appends remote request data to subsegment similar to what Subsegment.prototype.addRemoteRequestData
+ * does in core/lib/segments/attributes/subsegment.js
+ * @param {Subsegment} subsegment
+ * @param {Fetch Request} request
+ * @param {Fetch Request or null|undefined} response
+ * @param {boolean} downstreamXRayEnabled
+ */
+Subsegment.prototype.addFetchRequestData = function addFetchRequestData(request, response, downstreamXRayEnabled) {
+  this.http = {
+    request: {
+      url: request.url.toString(),
```

Comment:
```suggestion
      url: (request.url) ? request.url.toString() : '',
```

Reply:
Ok, done.  I also put in a non-set check for method.  .d.ts file is added for Subsegment.addFetchRequestData